src/creepy/subprocess/pypen.py

```diff
@@ -167,10 +167,35 @@ def _make_proxy_type(interface: str):
     def proxy_init(self, process):
         self._process = process
 
-    attrs['__init__'] = proxy_init
+    def proxy_enter(self):
+        self._process.__enter__()
+        return self
+
+    def proxy_reduce(self):
+        process_state = self._process.__getstate__()
+        return (_reconstruct_proxy, (interface, process_state))
+
+    attrs.update({
+        '__init__': proxy_init,
+        '__enter__': proxy_enter,
+        '__exit__': _delegate_to_process('__exit__'),
+        '__reduce__': proxy_reduce,
+        'request': _delegate_to_process('request'),
```

Comment:
Why add request? I think compiled object doesn't request anymore, all the compiled methods are the only methods left for you.